### PR TITLE
feat: v1beta1 CRP validation for PickAll Placement policy

### PR DESCRIPTION
**Pull request by Arvindthiru**
### Description of your changes

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/utils/validator/clusterresourceplacement_test.go [hunk 1]
```diff
@@ -175,6 +175,12 @@ func Test_validateClusterResourcePlacementAlpha(t *testing.T) {
 func Test_validateClusterResourcePlacement(t *testing.T) {
 	unavailablePeriodSeconds := -10
 	var numberOfClusters int32 = 1
+	resourceSelector := placementv1beta1.ClusterResourceSelector{
```
Comment (ryanzhang-oss) [inline pkg/utils/validator/clusterresourceplacement_test.go:178]: This one function test all is not easy to follow and maintain. There are going to be too many cases. Why not split them into UT? This is more like e2e.

File: test/e2e/webhook_test.go [hunk 2]
```diff
@@ -107,4 +108,40 @@ var _ = Describe("webhook tests for CRP UPDATE operations", Ordered, func() {
 			return nil
 		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
 	})
+
+	It("should deny update on CRP with invalid placement policy for PickAll", func() {
```
Comment (ryanzhang-oss) [inline test/e2e/webhook_test.go:112]: Strictly speaking we don't really need this e2e. Doesn't hurt to have one but why not pick something specific error related to the pickAll?

Comment (Arvindthiru) [reply test/e2e/webhook_test.go:112]: Agreed, we don't actually need the E2Es but I wanted to at least cover some important cases in E2Es that are already covered by UTs just to provide solid proof

I am testing a bad label selector within cluster affinity which is within RequiredDuringSchedulingIgnoredDuringExecution which is specific to PickAll, I also added topology constraints which is not allowed for PickAll

File: pkg/utils/validator/clusterresourceplacement_test.go [hunk 3]
```diff
@@ -511,17 +480,139 @@ func Test_validateClusterResourcePlacement(t *testing.T) {
...
+					ResourceSelectors: []placementv1beta1.ClusterResourceSelector{resourceSelector},
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType:    placementv1beta1.PickFixedPlacementType,
+						NumberOfClusters: &numberOfClusters,
+					},
+				},
+			},
+			wantErr: true,
+		},
+		"valid placement policy - PickAll with non nil affinity": {
```
[the remark this answers is not in the document]
Comment (Arvindthiru) [reply pkg/utils/validator/clusterresourceplacement_test.go:548]: It is allowed, I'm expecting no errors

File: pkg/utils/validator/clusterresourceplacement_test.go [hunk 4]
```diff
@@ -511,17 +480,139 @@ func Test_validateClusterResourcePlacement(t *testing.T) {
...
+		t.Run(testName, func(t *testing.T) {
+			ResourceInformer = testCase.resourceInformer
+			if err := ValidateClusterResourcePlacement(testCase.crp); (err != nil) != testCase.wantErr {
+				t.Errorf("ValidateClusterResourcePlacement_PickFixedPlacementPolicy() error = %v, wantErr %v", err, testCase.wantErr)
+			}
+		})
+	}
+}
+
+func Test_ValidateClusterResourcePlacement_PickAllPlacementPolicy(t *testing.T) {
```
Comment (ryanzhang-oss) [inline pkg/utils/validator/clusterresourceplacement_test.go:509]: is the invalid case of PreferreAffinity covered?

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**